packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -139,60 +223,79 @@ export class VaFileInput {
-                  {' '}
-                  {i18next.t('required')}
-                </span>
-              )}
-            </label>
-          )}
+          {label && this.renderLabelOrHeader(label, required, headerSize)}
           {hint && (
-            <span class="usa-hint" id="input-hint-message">
-              {hint}
-            </span>
+            <div class="usa-hint" id="input-hint-message">{hint}</div>
           )}
-          <slot></slot>
           <span id="input-error-message" role="alert">
-            {error && (
-              <Fragment>
-                <span class="usa-sr-only">{i18next.t('error')}</span>
-                <span class="usa-error-message">{error}</span>
-              </Fragment>
+             {error && (
+               <Fragment>
+                 <span class="usa-sr-only">{i18next.t('error')}</span>
+                 <span class="usa-error-message">{error}</span>
+               </Fragment>
+             )}
+           </span>
+          <div class="file-input-wrapper" onDrop={this.handleDrop}>
+            <input
+              id="fileInputField"
+              class="file-input"
+              style={{ visibility: this.uploadStatus === 'success' ? 'hidden' : 'unset' }}
+              type="file"
+              ref={el => this.fileInputRef = el as HTMLInputElement}
+              name={name}
+              accept={accept}
+              aria-describedby={ariaDescribedbyIds}
+              onChange={this.handleChange}
+            />
+            {uploadStatus === 'idle' && (
+              <div>
+                <div class="sr-only">No files selected.</div>
+                <div class={fileInputTargetClasses}>
+                  <div class="file-input-box"></div>
+                  <div class="file-input-instructions">
+                    <span class="file-input-drag-text">Drag files here or </span>
+                    <span class="file-input-choose-text">choose from folder</span>
+                  </div>
+                </div>
+              </div>
             )}
-          </span>
-          <input
-            id="fileInputField"
-            class={inputClass}
-            type="file"
-            name={name}
-            accept={accept}
-            aria-describedby={ariaDescribedbyIds}
-            onChange={this.handleChange}
-          />
+            {uploadStatus !== 'idle' && (
+              <div class="selected-files-wrapper">
+                <div class="selected-files-label">Selected files</div>
+                <va-card class="va-card">
+                  <div class="file-info-section">
+                    <va-icon icon="file_present" size={5} class="file-icon"></va-icon>
+                    <span class="file-label">{file.name}</span>
+                    <span class="file-size-label">{this.formatFileSize(file.size)}</span>
+                  </div>
+                  {(uploadStatus === 'success' || uploadStatus === 'failure') && file && (
```

Comment:
Do we need the `uploadStatus` check here? On first pass it seems like we could just check `file` and that would suffice.